nl/nl_linux.go

```diff
@@ -847,16 +895,16 @@ func (s *NetlinkSocket) Receive() ([]syscall.NetlinkMessage, *unix.SockaddrNetli
 
 // SetSendTimeout allows to set a send timeout on the socket
 func (s *NetlinkSocket) SetSendTimeout(timeout *unix.Timeval) error {
-	// Set a send timeout of SOCKET_SEND_TIMEOUT, this will allow the Send to periodically unblock and avoid that a routine
-	// remains stuck on a send on a closed fd
-	return unix.SetsockoptTimeval(int(s.fd), unix.SOL_SOCKET, unix.SO_SNDTIMEO, timeout)
+	duration := (time.Duration(timeout.Sec) * time.Second) + (time.Duration(timeout.Usec) * time.Microsecond)
+	atomic.StoreInt64(&s.sendTimeout, int64(duration))
```

Comment:
The `unix.Timeval` type already has a methods to convert to and from a `time.Duration` -- just without naming the type, [presumably because it's context-dependent whether a timeval represents a duration or timestamp.](https://go.dev/issues/2534#issuecomment-66060429)
```suggestion
	atomic.StoreInt64(&s.sendTimeout, timeout.Nano())
```

Reply:
Oh, that's handy! Thank you.